### PR TITLE
update to pass response body to gson

### PR DESCRIPTION
**Pull request by jeyalee**

### CODE REVIEW

File: src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt [hunk 1]
```diff
@@ -99,12 +100,12 @@ class BandcampAPIConnector constructor(private val bandcampUser: String, private
                                         .ignoreContentType(true)
                                         .timeout(timeout)
                                         .cookies(cookies)
+                                        .method(Method.POST)
                                         .requestBody("{\"fan_id\": $fanId, \"older_than_token\": \"$lastToken\"}")
-                                        .post()
+                                        .execute()
```
Comment (Ezwen) [inline src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt:105]: Sure why not but what does this change exactly? How is `method`+`execute` different from `post`?

File: src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt [hunk 2]
```diff
@@ -99,12 +100,12 @@ class BandcampAPIConnector constructor(private val bandcampUser: String, private
...
                             }, retries)
 
-
-                    val parsedCollectionData = gson.fromJson(theRest!!.wholeText(), ParsedCollectionItems::class.java)
+                    val parsedCollectionData = gson.fromJson(theRest!!.body(), ParsedCollectionItems::class.java)
```
Comment (Ezwen) [inline src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt:108]: I wonder, isn't it only this change here that fixed your problem?

**Review by Ezwen — COMMENTED** [review]
Thanks for the PR! I'm trying to better understand before the code change before the merge, so I've added a few questions, if you have time to answer them?

File: src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt (hunk 1 above)
Comment (jeyalee) [reply src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt:105]: So `post()` executes the request _and_ parses the result while `execute()` just executes the request. With my use case, jsoup was not parsing the result correctly. 

This is what `theRest!!.wholeText()` was returning before `gson.fromJson` executed.
```json
{"file":{"mp3-v0":"https://bandcamp.com/stream_redirect?enc=mp3-v0&track_id=29128347&ts=1600291760&t=c2360d89add21e40f7fe234fab22ac61bf9cda63"},"title":"増>>>>減<<<
```

Causing this error `com.google.gson.stream.MalformedJsonException: Unterminated string at line 1 column 95318 path $.tracklists.a2506905463[17].title`

File: src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt (hunk 2 above)
Comment (jeyalee) [reply src/main/kotlin/bandcampcollectiondownloader/BandcampAPIConnector.kt:108]: `post()` returns a parsed `Document` and  `execute()` returns a `Response`. The `body()` methods are different for each. The former returns an `Element` and the latter returns a plain string.

**Comment by Ezwen** [comment]
Perfect, all is clear now! Let's merge

**Pull request merged**